src/components/views/settings/encryption/ChangeRecoveryKey.tsx

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+import React, { FormEventHandler, JSX, MouseEventHandler, useState } from "react";
+import {
+    Breadcrumb,
+    Button,
+    ErrorMessage,
+    Field,
+    IconButton,
+    Label,
+    Root,
+    Text,
+    TextControl,
+} from "@vector-im/compound-web";
+import CopyIcon from "@vector-im/compound-design-tokens/assets/web/icons/copy";
+import { logger } from "matrix-js-sdk/src/logger";
+
+import { _t } from "../../../../languageHandler";
+import { EncryptionCard } from "./EncryptionCard";
+import { useMatrixClientContext } from "../../../../contexts/MatrixClientContext";
+import { useAsyncMemo } from "../../../../hooks/useAsyncMemo";
+import { copyPlaintext } from "../../../../utils/strings";
+import { withSecretStorageKeyCache } from "../../../../SecurityManager";
+
+/**
+ * The possible states of the component.
+ * - `inform_user`: The user is informed about the recovery key.
+ * - `save_key_setup_flow`: The user is asked to save the new recovery key during the setup flow.
+ * - `save_key_change_flow`: The user is asked to save the new recovery key during the change key flow.
+ * - `confirm`: The user is asked to confirm the new recovery key.
+ */
+type State = "inform_user" | "save_key_setup_flow" | "save_key_change_flow" | "confirm";
+
+interface ChangeRecoveryKeyProps {
+    /**
+     * If true, the component will display the flow to change the recovery key.
+     * If false,the component will display the flow to set up a new recovery key.
+     */
+    userHasKeyBackup: boolean;
+    /**
+     * Called when the recovery key is successfully changed.
+     */
+    onFinish: () => void;
+    /**
+     * Called when the cancel button is clicked or when we go back in the breadcrumbs.
+     */
+    onCancelClick: () => void;
+}
+
+/**
+ * A component to set up or change the recovery key.
+ */
+export function ChangeRecoveryKey({
+    userHasKeyBackup,
+    onFinish,
+    onCancelClick,
+}: ChangeRecoveryKeyProps): JSX.Element | null {
+    const matrixClient = useMatrixClientContext();
+
+    const [state, setState] = useState<State>(userHasKeyBackup ? "save_key_change_flow" : "inform_user");
+
+    // We create a new recovery key, the recovery key will be displayed to the user
+    const recoveryKey = useAsyncMemo(() => matrixClient.getCrypto()!.createRecoveryKeyFromPassphrase(), []);
+    if (!recoveryKey?.encodedPrivateKey) return null;
+
+    let content: JSX.Element;
+    switch (state) {
+        case "inform_user":
+            content = (
```

Comment:
```suggestion
            // Show a panel explaining what "recovery" is for, and what a recovery key does.
            content = (
```

Reply:
Can you add similar comments to the other two cases, giving a quick summary of what the user sees?